src/Simplify.elm

```diff
@@ -2876,6 +2911,154 @@ stringReverseChecks ({ parentRange, fnRange, firstArg } as checkInfo) =
                 checkInfo
 
 
+stringSliceChecks : CheckInfo -> List (Error {})
+stringSliceChecks checkInfo =
+    case ( checkInfo.firstArg, checkInfo.secondArg, checkInfo.thirdArg ) of
+        ( _, _, Just (Node _ (Expression.Literal "")) ) ->
+            [ Rule.errorWithFix
+                { message = "Using String.slice on an empty string will result in an empty string"
+                , details = [ "You can replace this call by an empty string." ]
+                }
+                checkInfo.fnRange
+                [ Fix.replaceRangeBy checkInfo.parentRange "\"\"" ]
+            ]
+
+        ( _, Just (Node _ (Expression.Integer 0)), _ ) ->
+            [ Rule.errorWithFix
+                { message = "Using String.slice with end index 0 will result in an empty string"
+                , details = [ "You can replace this call by an empty string." ]
+                }
+                checkInfo.fnRange
+                [ Fix.replaceRangeBy checkInfo.parentRange "always \"\"" ]
```

Comment:
This should probably be `""` if there is a third argument, and ``always ""`` if there isn't, no?

Reply:
no because the replaced range only ranged up to the second :)

Reply:
Thanks! I thought I had updated the range to only go up until including the last argument we know of... guess not, let me fix